telegram/start_telegram_bot.js

```diff
@@ -480,17 +488,26 @@ module.exports = (args, cbk) => {
           return cbk();
         }
 
-        inquirer.prompt([interaction.user_id_prompt]).then(({code}) => {
-          if (!code) {
-            return cbk([400, 'ExpectedNumericConnectCode']);
-          }
-
-          connectedId = code;
-
-          return cbk();
-        });
+        return ask({
+          type: 'input',
+          message: interaction.user_id_prompt.message,
+          name: interaction.user_id_prompt.name,
+          validate: code => {
+            if (!code || !isNumber(code)) {
+              return `Expected Numeric Connect Code`;
+            }
+            const key = args.key;
 
-        return;
+            if (key.startsWith(code)) {
```

Comment:
```suggestion
            if (key.startsWith(`${code}:`)) {
```

Maybe this would be better because in theory you could have a bot id like 12345 and a userid like 1234 although it's unlikely